src/vehicle.cpp

```diff
@@ -3193,6 +3216,42 @@ int vehicle::basic_consumption(const itype_id &ftype) const
     return fcon;
 }
 
+int vehicle::supplemental_consumption( const itype_id &ftype ) const
+{
+    int fcon = 0;
+    if( chainsaw_on ) {
+        std::vector<int> all_chainsaws = all_parts_with_feature( "CHAINSAW" );
+        for( size_t c = 0; c < all_chainsaws.size(); ++c ) {
```

Comment:
This loop *could* be written simpler as
```C++
for( auto &part : all_parts_with_feature( "CHAINSAW" ) ) {
    if( part_info( part ) ... ) {
```

Specifically: avoids the index variable by using a range-based for loop, avoid storing the result of `all_parts_with_feature` (it's not used except for the iteration).